andela/wg-gnext
# #154239888Ft Add Macrocycle, Mesocycle and Microcycle support

#### What does this PR do?
Add support for macrocycle, mesocycle and microcycle plans.

#### Description of Task to be completed?
Users should be able to create a fitness programs drafted more than only one week upto a year.

#### How should this be manually tested?
Login to `wger` app and on the navigation bar click `Workouts` dropdown and select `Workout schedules`.

When the page loads click on `add schedule` to create a new schedule, on the provided form ensure you check `use periodization` checkbox before saving the new schedule.

After saving, click `Add workout` button to add workout to the created schedule. A form will pop up with which you will be required to choose your desired workout and `duration` which is a dropdown containing three `cycles(macrocycle, mesocycle, microcycle)`, choose any of them and save the form.
You will be able to see the schedules and associated cycles listed.

#### What are the relevant pivotal tracker stories?
[https://www.pivotaltracker.com/story/show/154239888](https://www.pivotaltracker.com/story/show/154239888)

#### Screenshots (if appropriate)
<img width="1428" alt="screen shot 2018-02-08 at 11 25 55" src="https://user-images.githubusercontent.com/15877982/35962619-18234818-0cc3-11e8-82c5-c198e9391bdd.png">

<img width="1418" alt="screen shot 2018-02-08 at 11 26 34" src="https://user-images.githubusercontent.com/15877982/35962632-237df0aa-0cc3-11e8-9f7c-5199286c995b.png">

<img width="1440" alt="screen shot 2018-02-08 at 11 26 12" src="https://user-images.githubusercontent.com/15877982/35962628-1e0c2eb6-0cc3-11e8-9879-dd71ba7d946a.png">
